### DISCUSSION TITLE
Codai installation problems

### DISCUSSION THREAD

**Discussion by pkirkaas** [meysamhadeli/codai#95] [Q&A]
Codai sounds like a great project & I was interested to try it out, but I wasn't able to install it.

I am on the latest Windows 11 Pro, with go version go1.23.4 windows/amd64.

I tried to install codai as per instructions:

```sh
go install github.com/meysamhadeli/codai@latest
```

but got the error:

```sh
go: downloading github.com/xo/terminfo v0.0.0-20220910002029-abceb7e1c41e
# runtime/cgo
gcc_libinit_windows.c: In function '_cgo_beginthread':
gcc_libinit_windows.c:143:27: error: implicit declaration of function '_beginthread'; did you mean '_cgo_beginthread'? [-Werror=implicit-function-declaration]
  143 |                 thandle = _beginthread(func, 0, arg);
      |                           ^~~~~~~~~~~~
      |                           _cgo_beginthread
cc1: all warnings being treated as errors
```

So I started a `WSL` Linux terminal, installed `go version go1.23.4 linux/amd64`, and tried again to install `codai`, but got the error:

`go/pkg/mod/github.com/smacker/go-tree-sitter@v0.0.0-20240827094217-dd81d9e9be82/iter.go:17:18: undefined: Node`

Are you familiar with these issues? Do you have any suggestions? Are you still working on this interesting project?

Paul

**Comment by meysamhadeli** [comment]
> Codai sounds like a great project & I was interested to try it out, but I wasn't able to install it.
> 
> I am on the latest Windows 11 Pro, with go version go1.23.4 windows/amd64.
> 
> I tried to install codai as per instructions:
> 
> ```shell
> go install github.com/meysamhadeli/codai@latest
> ```
> 
> but got the error:
> 
> ```shell
> go: downloading github.com/xo/terminfo v0.0.0-20220910002029-abceb7e1c41e
> # runtime/cgo
> gcc_libinit_windows.c: In function '_cgo_beginthread':
> gcc_libinit_windows.c:143:27: error: implicit declaration of function '_beginthread'; did you mean '_cgo_beginthread'? [-Werror=implicit-function-declaration]
>   143 |                 thandle = _beginthread(func, 0, arg);
>       |                           ^~~~~~~~~~~~
>       |                           _cgo_beginthread
> cc1: all warnings being treated as errors
> ```
> 
> So I started a `WSL` Linux terminal, installed `go version go1.23.4 linux/amd64`, and tried again to install `codai`, but got the error:
> 
> `go/pkg/mod/github.com/smacker/go-tree-sitter@v0.0.0-20240827094217-dd81d9e9be82/iter.go:17:18: undefined: Node`
> 
> Are you familiar with these issues? Do you have any suggestions? Are you still working on this interesting project?
> 
> Paul

@pkirkaas  
Hi,
Thanks for your feedback :)
I updated all packages to latest version in new version [v1.8.1](https://github.com/meysamhadeli/codai/releases/tag/v1.8.1) and check it out again for installing codai in Windows 11 and go version [1.23.4](https://go.dev/dl/go1.23.4.windows-amd64.msi) and it work's properly and didn't have any error that you mentioned.
- Firstly please delete codai from installation path in windows `C:\Users\TestUser\go\bin\codai.exe`.
- After that inistall codai again with the new latest version `go install github.com/meysamhadeli/codai@latest` and everything should work properly.
